pkg/apis/batch/validation/validation.go

```diff
@@ -162,6 +163,13 @@ func ValidateCronJob(scheduledJob *batch.CronJob) field.ErrorList {
 	// CronJobs and rcs have the same name validation
 	allErrs := apivalidation.ValidateObjectMeta(&scheduledJob.ObjectMeta, true, apivalidation.ValidateReplicationControllerName, field.NewPath("metadata"))
 	allErrs = append(allErrs, ValidateCronJobSpec(&scheduledJob.Spec, field.NewPath("spec"))...)
+	if len(scheduledJob.ObjectMeta.Name) > apimachineryvalidation.DNS1035LabelMaxLength-11 {
```

Comment:
Job creates Pods which have further suffixes, right?  And Pod names also are limited in size.

Reply:
Pod names' size limits are 255 characters, like most Kubernetes resources. As far as I can tell only job names have a size limit of 63 characters (because job names get copied in to a label, which have a lower size limit than Kubernetes resource names).